### PR TITLE
Apply patches to properly load bunyan config file

### PR DESCRIPTION
**Pull request by christinebrass**
The Bunyan plugin was not actually loading in an app's config file, therefore always using a default Bunyan configuration.

Changes made:
* Use requireModule to digest bunyan config file
* Update test to mock requireModule
* Add test for empty config case

### CODE REVIEW

File: packages/gluestick-plugin-bunyan/src/server.js [hunk 1]
```diff
@@ -19,24 +20,41 @@ export type Logger = {
...
   name: 'default name',
 };
 
-const bunyanPlugin = (): { logger: Logger } => {
-  let options: Options = defaultSettings;
+const bunyanPlugin = (
+  opts: Object,
+  { requireModule, defaultLogger }: PluginUtilities,
+): { logger?: Logger } => {
+  let options: Options = { ...defaultSettings };
```
Comment (threehams) [inline packages/gluestick-plugin-bunyan/src/server.js]: Is there any reason to make a shallow copy here? Nothing in this file is changing it.

File: packages/gluestick-plugin-bunyan/src/server.js [hunk 2]
```diff
@@ -19,24 +20,41 @@ export type Logger = {
...
+};
+
 const defaultSettings: Options = {
   name: 'default name',
 };
 
-const bunyanPlugin = (): { logger: Logger } => {
-  let options: Options = defaultSettings;
+const bunyanPlugin = (
+  opts: Object,
```
Comment (threehams) [inline packages/gluestick-plugin-bunyan/src/server.js]: This seems to be unused - is it just here by convention? Maybe it could be typed to `null`.

File: packages/gluestick-plugin-bunyan/src/server.js [hunk 3]
```diff
@@ -19,24 +20,41 @@ export type Logger = {
...
+  { requireModule, defaultLogger }: PluginUtilities,
+): { logger?: Logger } => {
+  let options: Options = { ...defaultSettings };
   try {
-    // $FlowIgnore
-    const userSettings: Options = require('src/bunyan.config.js').default;
+    const userSettings: Options = requireModule(
+      path.join(process.cwd(), 'src/bunyan.config.js'),
```
Comment (threehams) [inline packages/gluestick-plugin-bunyan/src/server.js]: I've gotten to like the aliases because of the automatic typing it gives us, though there's no reason to change this right now.

File: packages/gluestick-plugin-bunyan/src/server.js [hunk 4]
```diff
@@ -19,19 +20,29 @@ export type Logger = {
...
 };
 
-const bunyanPlugin = (): { logger: Logger } => {
-  let options: Options = defaultSettings;
+type BunyanPlugin = (
+  opts: Object,
+  pluginUtilities: PluginUtilities,
+) => { logger?: Logger };
+
+const bunyanPlugin: BunyanPlugin = (opts, { requireModule }) => {
```
Comment (threehams) [inline packages/gluestick-plugin-bunyan/src/server.js:36]: I wonder if the future of plugins should include some first-class configuration stuff.

**Review by threehams — APPROVED** [review]
